src/Shared/runtime/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -0,0 +1,361 @@
+                    nativeRegistration.StreamClose);
+            StreamStartDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.StreamStartDelegate>(
+                    nativeRegistration.StreamStart);
+            StreamShutdownDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.StreamShutdownDelegate>(
+                    nativeRegistration.StreamShutdown);
+            StreamSendDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.StreamSendDelegate>(
+                    nativeRegistration.StreamSend);
+            StreamReceiveCompleteDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.StreamReceiveCompleteDelegate>(
+                    nativeRegistration.StreamReceiveComplete);
+            StreamReceiveSetEnabledDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.StreamReceiveSetEnabledDelegate>(
+                    nativeRegistration.StreamReceiveSetEnabled);
+            SetContextDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.SetContextDelegate>(
+                    nativeRegistration.SetContext);
+            GetContextDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.GetContextDelegate>(
+                    nativeRegistration.GetContext);
+            SetCallbackHandlerDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.SetCallbackHandlerDelegate>(
+                    nativeRegistration.SetCallbackHandler);
+
+            SetParamDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.SetParamDelegate>(
+                    nativeRegistration.SetParam);
+            GetParamDelegate =
+                Marshal.GetDelegateForFunctionPointer<MsQuicNativeMethods.GetParamDelegate>(
+                    nativeRegistration.GetParam);
+
+            RegistrationOpenDelegate(Encoding.UTF8.GetBytes("SystemNetQuic"), out IntPtr ctx);
+            _registrationContext = ctx;
+        }
+
+        internal static MsQuicApi Api { get; }
+
+        internal static bool IsQuicSupported { get; }
+
+        static MsQuicApi()
+        {
+            // MsQuicOpen will succeed even if the platform will not support it. It will then fail with unspecified
+            // platform-specific errors in subsequent callbacks. For now, check for the minimum build we've tested it on.
+
+            // TODO:
+            // - Hopefully, MsQuicOpen will perform this check for us and give us a consistent error code.
+            // - Otherwise, dial this in to reflect actual minimum requirements and add some sort of platform
+            //   error code mapping when creating exceptions.
+
+            OperatingSystem ver = Environment.OSVersion;
+
+            if (ver.Platform == PlatformID.Win32NT && ver.Version < new Version(10, 0, 19041, 0))
```

Comment:
This check fails in our repo.

Reply:
@scalablecory this is something we'll need to revisit. Environment.OSVersion is extremely unreliable and shouldn't be used in production code. I'll file an issue.

Reply:
https://github.com/dotnet/runtime/issues/32575